### PR TITLE
Add validating webhook

### PR DESCRIPTION
**Pull request by yuvalturg**
**Release note**:

```release-note
Add a validating webhook for HCO
```

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by orenc1** [comment]
Hi @yuvalturg 
Note that the HCO pod is CrashLooping with:
`"Manager exited non-zero","error":"open /apiserver.local.config/certificates/apiserver.crt: no such file or directory"`

**Comment by yuvalturg** [comment]
> Hi @yuvalturg
> Note that the HCO pod is CrashLooping with:
> `"Manager exited non-zero","error":"open /apiserver.local.config/certificates/apiserver.crt: no such file or directory"`

Yes, thanks, this is mounted by OLM, i'll add a check

**Comment by yuvalturg** [comment]
> Yuval,
> can you please rebase this on current master and run again hack/build-manifests.sh to regenerate CSV v1.2.0 and not anymore v1.1.0?

Done

**Comment by tiraboschi** [comment]
> Done

Please revert all the changes to deploy/olm-catalog/kubevirt-hyperconverged/1.1.0/* because we want to have this only in v1.2.0 now

File: pkg/apis/hco/v1alpha1/hyperconverged_webhook.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	}
+
+	// The OLM limits the webhook scope to the namespaces that are defined in the OperatorGroup,
+	// by setting namespaceSelector in the ValidatingWebhookConfiguration.  We would like our webhook to intercept
+	// CREATE from all namespaces, and fail them if they're not in the correct namespace for HCO.
+	// To do that, we need to find our ValidatingWebhookConfiguration and reset the namespaceSelector that was
+	// defined by OLM.  Luckily, the OLM does not reconcile the ValidatingWebhookConfiguration
+
+	vwcList := &admissionregistrationv1.ValidatingWebhookConfigurationList{}
+	err := mgr.GetAPIReader().List(context.TODO(), vwcList, client.MatchingLabels{"olm.webhook-description-generate-name": hcoutil.HcoValidatingWebhook})
```
Comment (nunnatsa) [inline pkg/apis/hco/v1alpha1/hyperconverged_webhook.go:45]: context should represent a single flow; e.g. a specific HTTP call. Please consider to use the same context in all the calls - create a context at the beginning of the function and use it in all the calls to K8s.

Comment (yuvalturg) [reply pkg/apis/hco/v1alpha1/hyperconverged_webhook.go]: Done

File: deploy/olm-catalog/kubevirt-hyperconverged/1.1.0/hco00.crd.yaml [hunk 2]
```diff
@@ -27,9 +27,8 @@ spec:
             name:
               pattern: kubevirt-hyperconverged
               type: string
-            namespace:
```
Comment (nunnatsa) [inline deploy/olm-catalog/kubevirt-hyperconverged/1.1.0/hco00.crd.yaml:30]: I think 1.1.0 should not be changed.

**Pull request merged**